CMakeLists.txt

```diff
@@ -43,6 +43,7 @@ option(IREE_BUILD_TENSORFLOW_ALL "Builds all TensorFlow compiler frontends." OFF
 option(IREE_BUILD_TENSORFLOW_COMPILER "Builds TensorFlow compiler frontend." OFF)
 option(IREE_BUILD_TFLITE_COMPILER "Builds the TFLite compiler frontend." OFF)
 option(IREE_BUILD_XLA_COMPILER "Builds TensorFlow XLA compiler frontend." OFF)
+option(IREE_BUILD_BAREMETAL "Builds IREE in bare-metal configuration." OFF)
```

Comment:
As this is, it isn't super clear to me what the "bare-metal" configuration turns on/off and if it is compatible with other options. My impression of "bare-metal" is that it is also a flexible label that could be applied to different systems with a variety of strict requirements, and the settings we choose now for it might not apply to all devices in that category.

Rather than define a course option like this, I think I'd rather have a collection of more specific options like `IREE_ENABLE_THREADING` and `IREE_ENABLE_NATIVE_MODULES`. Then we could provide toolchain files like the existing `build_tools/cmake/riscv.toolchain.cmake` that set subsets of those options based on platform requirements. We could also run on systems that _do_ support those things with the reduced feature set if we wanted to.

If we split the options then checks in non-root files or around specific targets would be easier to understand (e.g. don't build `event_pool` if threading is disabled).

Reply:
Great feedback! I wasn't sure about the naming. `IREE_ENABLE_THREADING` is definitely more clear and fit into the purpose of this flag.